assets/src/domain/eventEditor/services/apollo/mutations/prices/usePriceMutationHandler.ts

```diff
@@ -31,42 +32,41 @@ const usePriceMutationHandler = (): MutationHandler => {
 		};
 	};
 
-	const mutator = (mutationType: MutationType, input: MutationInput): MutatorGeneratedObject => {
-		// `ticketId` must be passed for price creation
-		// so as to properly set the relations
-		const { ticketId, ...restInput } = input;
-		const { priceType: priceTypeId } = input;
-		const variables = createVariables(mutationType, restInput);
-		/**
-		 * @todo update optimisticResponse
-		 */
-		let optimisticResponse: any;
+	const mutator = useCallback<MutationHandler>(
+		(mutationType, input) => {
+			// `ticketId` must be passed for price creation
+			// so as to properly set the relations
+			const { ticketId, ...restInput } = input;
+			const { priceType: priceTypeId } = input;
+			const variables = createVariables(mutationType, restInput);
 
-		const onUpdate = ({ proxy, entity: price }: OnUpdateFnOptions<Price>): void => {
-			// Read the existing data from cache.
-			let data: PricesList;
-			try {
-				data = proxy.readQuery<PricesList>(options);
-			} catch (error) {
-				data = null;
-			}
-			const prices = pathOr<PriceEdge>(DEFAULT_LIST_DATA, ['espressoPrices'], data);
+			const onUpdate = ({ proxy, entity: price }: OnUpdateFnOptions<Price>): void => {
+				// Read the existing data from cache.
+				let data: PricesList;
+				try {
+					data = proxy.readQuery<PricesList>(options);
+				} catch (error) {
+					data = null;
+				}
+				const prices = pathOr<PriceEdge>(DEFAULT_LIST_DATA, ['espressoPrices'], data);
 
-			switch (mutationType) {
-				case MutationType.Create:
-					onCreatePrice({ proxy, prices, price, ticketId, priceTypeId });
-					break;
-				case MutationType.Update:
-					onUpdatePrice({ price, priceTypeId });
-					break;
-				case MutationType.Delete:
-					onDeletePrice({ proxy, prices, price, deletePermanently: input?.deletePermanently });
-					break;
-			}
-		};
+				switch (mutationType) {
+					case MutationType.Create:
+						onCreatePrice({ proxy, prices, price, ticketId, priceTypeId });
+						break;
+					case MutationType.Update:
+						onUpdatePrice({ price, priceTypeId });
+						break;
+					case MutationType.Delete:
+						onDeletePrice({ proxy, prices, price, deletePermanently: input?.deletePermanently });
+						break;
+				}
+			};
 
-		return { variables, optimisticResponse, onUpdate };
-	};
+			return { variables, onUpdate };
+		},
+		[options.variables]
```

Comment:
shouldn't `variables` also be included here?

Reply:
![image](https://user-images.githubusercontent.com/1477453/80708456-5de28100-8af4-11ea-82e1-203be12378ac.png)